File: 0150 - Evaluate Reverse Polish Notation/Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int, vector<int>> stack;
        for (const string& token : tokens) {
            if (token == "+") {
                int second = stack.top();
                stack.pop();
                stack.top() += second;
            } else if (token == "-") {
                int second = stack.top();
                stack.pop();
                stack.top() -= second;
            } else if (token == "*") {
                int second = stack.top();
                stack.pop();
                stack.top() *= second;
            } else if (token == "/") {
                int second = stack.top();
                stack.pop();
                stack.top() /= second;
            } else {
                stack.push(stoi(token));
            }
        }
        return stack.top();
    }
};

File: 0150 - Evaluate Reverse Polish Notation/Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token == '+':
                second = stack[-1]
                stack.pop()
                stack[-1] += second
            elif token == '-':
                second = stack[-1]
                stack.pop()
                stack[-1] -= second
            elif token == '*':
                second = stack[-1]
                stack.pop()
                stack[-1] *= second
            elif token == '/':
                second = stack[-1]
                stack.pop()
                stack[-1] = int(stack[-1] / second)
            else:
                stack.append(int(token))
        return stack[-1]

